### ISSUE TITLE
data returned

### ISSUE THREAD

**Issue by k3vLar** [danielkrainas/node-netstat#26]
any particular reason the data doesnt return in a standardformat like JSON or just an array?  What is this?

{ protocol: 'tcp',
  local: { port: 24801, address: '127.0.0.1' },
  remote: { port: 49669, address: '127.0.0.1' },
  state: 'ESTABLISHED',
  pid: 3968 }
{ protocol: 'tcp',
  local: { port: 49669, address: '127.0.0.1' },
  remote: { port: 24801, address: '127.0.0.1' },
  state: 'ESTABLISHED',
  pid: 4660 }
{ protocol: 'tcp',
  local: { port: 49810, address: '127.0.0.1' },
  remote: { port: 49811, address: '127.0.0.1' },
  state: 'ESTABLISHED',
  pid: 11176 }
{ protocol: 'tcp',
  local: { port: 49811, address: '127.0.0.1' },
  remote: { port: 49810, address: '127.0.0.1' },
  state: 'ESTABLISHED',
  pid: 11176 }
{ protocol: 'tcp',
  local: { port: 49822, address: '127.0.0.1' },
  remote: { port: 49823, address: '127.0.0.1' },
  state: 'ESTABLISHED',
  pid: 10540 }

**Comment by danielkrainas** [comment]
Hi @k3vLar ! That appears to be the data object output in the console. What do you think it's returning? The data is returned as a __standard__ JavaScript object; there is no magic, classing, prototypes, or otherwise involved. The handler is called once for every parsed line of data returned from netstat, not just once for the aggregate result.

```
var netstat = require('node-netstat');

const allResults = [];
netstat({
	filter: {
		pid: 4123,
		protocol: 'tcp'
	},
	limit: 5
}, function (entryData) {
    // access fields from parsed results
    entryData.protocol;
    entryData.remote.port;

    // add to aggregate result collection
    allResults.push(entryData);
});
```

I'm happy to help but I do feel there's a misunderstanding on your end as to how this library is used. I'd suggest checking out the README and if you are still having problems, to come back and we can work through them.

**Comment by k3vLar** [comment]
SMH i feel like such a nub HAHA, thanks works perfectly as i need it now.

**Comment by danielkrainas** [comment]
Happy to hear it!

**Comment by k3vl4rAtAirside** [comment]
@danielkrainas , any chance you can help me again man?  this suddenly stopped returning any netstat data a few weeks ago.

edit: ah nevermind, this work with mojave? :(

even the below returns nothing at all:
var netstat = require('node-netstat');

netstat({
    filter: {
    },
    limit: 5
}, function (data) {
        console.log(data);
});

**Comment by danielkrainas** [comment]
@k3vl4rAtAirside I don't know if it's ever been tested on Mojave, I know it was tested on Yosemite. You could try manually running the `netstat` command to see what mojave outputs `netstat -v -n -p tcp`

**Comment by k3vl4rAtAirside** [comment]
@danielkrainas yeah netstat works on mojave, conflict is somewhere else.

On mojave I'm just running it manually and splitting the data up now.

https://github.com/k3vl4rAtAirside/netglobe/blob/master/js/preload2.js

`Kevins-MBP:k3vl4r k3vl4r$ netstat -v -n -p tcp
Active Internet connections
Proto Recv-Q Send-Q  Local Address          Foreign Address        (state)     rhiwat shiwat    pid   epid  state    options
tcp4       0      0  192.168.1.191.52044    31.13.66.35.443        ESTABLISHED 131072 131860   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52043    31.13.66.19.443        ESTABLISHED 131072 131860   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52041    140.82.114.26.443      ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp4      24      0  192.168.1.191.52040    140.82.114.4.443       CLOSE_WAIT  131072 132432   9811      0 0x0122 0x00000008
tcp4       0      0  192.168.1.191.52039    140.82.114.4.443       ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp4      24      0  192.168.1.191.52036    140.82.114.4.443       CLOSE_WAIT  131072 132432   9811      0 0x0122 0x00000008
tcp4       0      0  192.168.1.191.52032    151.101.201.7.443      ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52029    192.30.253.125.443     ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52027    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52026    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52025    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52024    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52023    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52022    151.101.200.133.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52021    185.199.110.154.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52013    151.101.201.7.443      ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.52012    151.101.201.7.443      ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51994    172.217.7.197.443      ESTABLISHED 131072 131328   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51983    151.101.200.106.443    ESTABLISHED 1034264 131480    333      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51980    99.86.191.154.443      ESTABLISHED 131072 131768    333      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51979    151.101.200.106.443    ESTABLISHED 131072 131480    333      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51976    34.194.201.2.443       ESTABLISHED 131072 131768    333      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51941    162.247.242.21.443     ESTABLISHED 131072 131860    511      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51939    151.101.202.110.443    ESTABLISHED 131072 131480    511      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51930    52.1.197.9.443         ESTABLISHED 131072 131072    511      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51921    18.234.32.173.443      ESTABLISHED 157136 131072    511      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51919    151.101.201.140.443    ESTABLISHED 131072 131480   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51825    162.159.134.234.443    ESTABLISHED 262144 131400    330      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51817    140.82.113.25.443      ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51812    140.82.113.25.443      ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51809    192.168.1.244.7678     ESTABLISHED 131768 131768    511      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51806    54.210.55.106.443      ESTABLISHED 131072 131768     80      0 0x0182 0x00020000
tcp4       0      0  192.168.1.191.51801    31.13.66.4.443         ESTABLISHED 131072 131072   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51799    162.159.130.233.443    ESTABLISHED 262144 131400    330      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51796    140.82.113.25.443      ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp6       0      0  fe80::aede:48ff:.51791 fe80::aede:48ff:.49255 ESTABLISHED 131072 131376    259      0 0x0182 0x00000000
tcp6       0      0  fe80::aede:48ff:.51790 fe80::aede:48ff:.49251 ESTABLISHED 131136 131376    475      0 0x0182 0x00000000
tcp6       0      0  fe80::aede:48ff:.51789 fe80::aede:48ff:.49241 ESTABLISHED 131092 131376  10871      0 0x0182 0x00000000
tcp4       0      0  192.168.1.191.51782    31.13.66.4.443         ESTABLISHED 131072 131860   9811      0 0x0102 0x00000008
tcp4       0      0  192.168.1.191.51781    140.82.113.25.443      ESTABLISHED 131072 132432   9811      0 0x0102 0x00000008
tcp6       0      0  fe80::aede:48ff:.49156 fe80::aede:48ff:.49261 ESTABLISHED 131072 131376    201      0 0x0182 0x00000004
tcp6       0      0  fe80::aede:48ff:.51778 fe80::aede:48ff:.59602 ESTABLISHED 131072 131376     41      0 0x0182 0x00000000
tcp4       0      0  192.168.1.191.51777    17.57.144.53.5223      ESTABLISHED 131072 131072     79      0 0x0182 0x00020000
tcp6       0      0  fe80::aede:48ff:.50338 fe80::aede:48ff:.49220 ESTABLISHED 131072 131376    371      0 0x0182 0x00000000
tcp6       0      0  fe80::aede:48ff:.49156 fe80::aede:48ff:.49239 ESTABLISHED 131072 131376    201      0 0x0182 0x00000004
tcp6       0      0  fe80::aede:48ff:.49838 fe80::aede:48ff:.49196 ESTABLISHED 131229 131376    201      0 0x0182 0x00000000
tcp4       0      0  127.0.0.1.4244         127.0.0.1.49257        ESTABLISHED 450632 146988     43      0 0x0182 0x00000004
tcp4       0      0  127.0.0.1.49257        127.0.0.1.4244         ESTABLISHED 400988 146988    484      0 0x0102 0x00000008
tcp6       0      0  fe80::aede:48ff:.49249 fe80::aede:48ff:.49216 ESTABLISHED 131072 131376    550      0 0x0182 0x00000000
tcp6       0      0  fe80::aede:48ff:.49160 fe80::aede:48ff:.49213 ESTABLISHED 131120 131376    226      0 0x0182 0x00000000
tcp6       0      0  fe80::aede:48ff:.49157 fe80::aede:48ff:.49200 ESTABLISHED 131229 131072     60      0 0x0182 0x00000000
Kevins-MBP:k3vl4r k3vl4r$`

**Comment by danielkrainas** [comment]
@k3vLar it looks like mojave is adding more to the output than the current darwin parser expects. The current parser expects 10 parts and mojave's output is 12, I will update the darwin parser factory tonight to hopefully address it. 

current parser: https://github.com/danielkrainas/node-netstat/blob/master/lib/parser-factories.js#L42

```
Proto Recv-Q Send-Q Local Address Foreign Address (state) rhiwat shiwat pid epid state options
tcp4 0 0 192.168.1.191.52044 31.13.66.35.443 ESTABLISHED 131072 131860 9811 0 0x0102 0x00000008
```

**Comment by danielkrainas** [comment]
tracking on #29

**Comment by danielkrainas** [comment]
@k3vLar I've published an update in [v1.8.0](https://github.com/danielkrainas/node-netstat/releases/tag/v1.8.0) that handles the additional output from mojave's `netstat`. Let me know how that works for you.
